### PR TITLE
Add plugin for customizing diffuse light color.

### PR DESCRIPTION
**Pull request by kircher1**
This is useful for rendering techniques where light color may change dynamically based on the position of the fragment being shaded (e.g. applying sun light to a large terrain)

### CODE REVIEW

**Comment by kircher1** [comment]
After a successful CI build, I will share a playground that ensures this functionality works as expected.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15511/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15511/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15511/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15511/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15511/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15511/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15511/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15511/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15511/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15511/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/ShadersWGSL/ShadersInclude/lightFragment.fx [hunk 1]
```diff
@@ -59,19 +63,19 @@
 
             // Diffuse contribution
             #ifdef HEMILIGHT{X}
-                info.diffuse = computeHemisphericDiffuseLighting(preInfo, light{X}.vLightDiffuse.rgb, light{X}.vLightGround);
+                info.diffuse = computeHemisphericDiffuseLighting(preInfo, diffuse.rgb, light{X}.vLightGround);
```
Comment (Popov72) [inline packages/dev/core/src/ShadersWGSL/ShadersInclude/lightFragment.fx:66]: `{X}` is missing for all `diffuse` occurrences.

**Review by Popov72 — CHANGES_REQUESTED** [review]

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15511/merge/testResults/webgpuplaywright/index.html

**Review by Popov72 — APPROVED** [review]

**Review by sebavan — APPROVED** [review]

**Pull request merged**